src/blockdata/transaction.rs

```diff
@@ -272,6 +272,11 @@ impl Transaction {
             Err(script::Error::SerializationError)
         }
     }
+
+    /// is this a coin base transaction?
+    pub fn is_coin_base (&self) -> bool {
+        self.input.len() == 1 && self.input [0].prev_hash == Sha256dHash::default()
```

Comment:
Doesnt this also need a nSequence == -1 test?

Reply:
I followed this: https://github.com/bitcoin/bitcoin/blob/3c2a41a9fc3cffe59cd5ad324efaf8418b53a128/src/primitives/transaction.h#L333

Reply:
Thanks a lot @TheBlueMatt . I was fooled by missing outpoint here. Introducing outpoint would make sense also because it has its own serialization for some other purposes like adding to filters or cacheing as coin keys. Pushed a correcting commit.